gui/wxpython/gui_core/pyedit.py

```diff
@@ -598,6 +626,12 @@ def __init__(self, parent, giface, id=wx.ID_ANY,
         if sys.platform != 'darwin':
             self.SetToolBar(self.toolbar)
 
+        # Get first (File) menu
+        menu = self.menubar.GetMenu(0)
+        self.recent_files = RecentFilesMenu(
+            app_name='pyedit', parent_menu=menu, pos=1,
+        )
+
         self.panel = PyStc(parent=self)
         self.controller = PyEditController(
             panel=self.panel, guiparent=self, giface=giface)
```

Comment:
Passing the recent_files as part of guiparent which is meant to be used as a parent for the GUI objects is sub-optimal. If the PyEditController really needs access to recent_files (it seems to me it does, but you have a better idea of what is needed), PyEditController should just take it as a parameter.

It is kind of hard to say what should be in frame and what in controller. wxPython sort of pushes you to put everything in the frame which results in bad code (just look elsewhere). Controller is an effort to have logic outside of the wxPython class (i.e, the frame).